File: 백준/Silver/7568. 덩치/덩치.py
import sys
input = sys.stdin.readline

n = int(input())
arr = []
rank = [1] * n

for i in range(n):
  keys = list(map(int, input().split()))
  arr.append(keys)

for i in range(n - 1):
  for j in range(i + 1, n):
    if (arr[i][0] < arr[j][0] and arr[i][1] < arr[j][1]):
      rank[i] += 1
    if (arr[i][0] > arr[j][0] and arr[i][1] > arr[j][1]):
      rank[j] += 1
      
  
for i in range(n):
  print(rank[i], end = ' ')
print()

File: 백준/Silver/7568. 덩치/덩치.js
/** @format */
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split(/\r?\n/);

function solution(input) {
    const N = Number(input[0]);
    const arr = [];
    const ans = [];

    for (let i = 0; i < N; i++) {
        const [kg, cm] = input[1 + i].split(" ").map(Number);
        ans.push(1);
        arr.push({kg: kg, cm: cm});
    }

    for (let i = 0; i < N; i++) {
        for (let j = 0; j < N; j++) {
            if (arr[i].kg < arr[j].kg && arr[i].cm < arr[j].cm) ans[i]++;
        }
    }

    let tmp = "";
    ans.forEach((e) => {
        tmp += `${e} `;
    });
    console.log(tmp);
}

solution(input);